src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/CaseController.java

```diff
@@ -66,8 +69,13 @@ public CaseData getCase(
         } else {
             throw (new RuntimeException(INVALID_CLIENT));
         }
-        return objectMapper.convertValue(caseDetails.getData(), CaseData.class)
-            .toBuilder().id(caseDetails.getId()).build();
+        if (null == caseDetails) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "case not found");
+        } else {
+            CaseData caseData = CaseUtils.getCaseData(caseDetails, objectMapper);
+            return caseData.toBuilder().noOfDaysRemainingToSubmitCase(
+                CaseUtils.getRemainingDaysSubmitCase(caseData)).build();
+        }
```

Comment:
move this block (72-78) within isAuthorised method.

Reply:
Done